db/db.js

```diff
@@ -44,40 +40,24 @@ const DBUtils = {
     }
   },
 
-  async _getSha1EntriesFromPrefixAndDo(sha1Prefix, aFoundCallback, aNotFoundCallback) {
-    // Only accept 6-character hash prefixes so requests:
-    // 1. can't get more than the intended hash range results FROM us
-    // 2. can't reveal a more specific hash query TO us
-    if (sha1Prefix.length !== 6) {
-      return await aNotFoundCallback();
-    }
-    const existingEntries = await EmailHash.query().where("sha1", "like", sha1Prefix + "%").eager("breaches");
-
-    if (existingEntries.length && aFoundCallback) {
-      return await aFoundCallback(existingEntries);
-    }
-
-    if (!existingEntries.length && aNotFoundCallback) {
-      return await aNotFoundCallback();
-    }
-  },
-
   // Used internally.
   async _addEmailHash(sha1, email, verified = false) {
     return await this._getSha1EntryAndDo(sha1, async aEntry => {
       // Entry existed, patch the email value if supplied.
       if (email) {
-        return await aEntry
-          .$query()
-          .patch({ email, verified })
-          .returning("*"); // Postgres trick to return the updated row as model.
+        const res = await knex("subscribers")
+          .update({email: email, verified: verified})
+          .where("id", "=", aEntry.id)
+          .returning("*");
+        return res[0];
       }
 
       return aEntry;
     }, async () => {
-      return await EmailHash
-        .query()
-        .insert({ sha1, email, verified });
+      const res = await knex("subscribers")
+        .insert({ sha1: sha1, email: email, verified: verified })
```

Comment:
Not sure if it is any easier just to use the shorthand syntax of `.insert({ sha1, email, verified })` since the keys and values are the "same".

There's probably an ESLint rule for that, if you want me to research it and address in a separate PR.

Reply:
Thought I had that before and it wasn't working. But it works now, so yup - added commit to use shorthand syntax.